Src/PerformanceCollector/Filtering/CollectionConfiguration.cs

```diff
@@ -0,0 +1,327 @@
+﻿namespace Microsoft.ApplicationInsights.Extensibility.Filtering
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Globalization;
+    using System.Linq;
+
+    using Microsoft.ApplicationInsights.DataContracts;
+    using Microsoft.ApplicationInsights.Extensibility.PerfCounterCollector.Implementation.QuickPulse.Helpers;
+
+    internal class CollectionConfiguration
+    {
+        private readonly CollectionConfigurationInfo info;
+
+        private readonly List<OperationalizedMetric<RequestTelemetry>> requestTelemetryMetrics = new List<OperationalizedMetric<RequestTelemetry>>();
+
+        private readonly List<OperationalizedMetric<DependencyTelemetry>> dependencyTelemetryMetrics =
```

Comment:
These are used on the perf-cricital path inside QuickPulseTelemetryProcessor. I'm fighting for speed there, so I only determine the actual telemetry item type once and then use if's and generic methods to process. We're not likely to have many more telemetry types, so I think it's worth the effort.